lms/djangoapps/grades/signals/handlers.py

```diff
@@ -123,7 +125,23 @@ def score_published_handler(sender, block, user, raw_earned, raw_possible, only_
                 )
 
     if update_score:
+        course = get_course(block.course_id)
+        original_course_grade = CourseGradeFactory().create(user, course)
+
+        # Set the problem score in CSM.
         score_modified_time = set_score(user.id, block.location, raw_earned, raw_possible)
+
+        course_overview = CourseOverview.objects.filter(id=block.course_id).first()
+        course_grade = CourseGradeFactory().create(user, course)
+
+        grade_changed = original_course_grade.percent != course_grade.percent
+
+        if original_course_grade.percent <= course_overview.lowest_passing_grade and grade_changed:
```

Comment:
If `grade_changed ` makes code more readable than it would be better to add another variable for `original_course_grade.percent <= course_overview.lowest_passing_grade`.

Reply:
Saying 'more readable' I mean that removing `grade_changed` variable will make `if` condition too long and so less readable, but yes, requested changes might make code better) I'll update it.